### DISCUSSION TITLE
Configuration in WSL Home Folder

### DISCUSSION THREAD

**Discussion by n-sweep** [wezterm/wezterm#2400] [Q&A] [answered]
I keep all my other configuration files in a dotfiles repo and stow it into my home folder on the Linux side, is it possible to get wezterm to recognize configuration files in the WSL home folder rather than (or in addition to) the Windows home folder? Do I just require the file with an absolute path or is there more appropriate way?

**Comment by wez** [comment, marked as answer]
The WSL filesystem isn't directly visible on the Windows host, so that isn't possible.

An option to consider is to move your WSL homedir to point to your windows home dir (or some sub-dir) so that it is `/mnt/c/Users/username` rather than `/home/username`; you can use something like `usermod` for that.

That way your WSL is using the same data as your windows host, rather than the other way around.

What I do is a bit different: my dotfiles repo has an "install" script that makes symlinks from the home dir into the files in the dotfiles repo.  I have that repo checked out in a location like `%HOME%/dotfiles` and it populates my windows home dir when run on windows, and does a deeper setup with things like the `.config` dir when run on unix systems (it's visible in my WSL instance at `/mnt/c/Users/wez/dotfiles` and sets up `~/.wezterm.lua` -> `/mnt/c/Users/wez/dotfiles/wezterm.lua`, and so on).

My dotfiles repo isn't shared publicly, sorry!

**Reply by n-sweep** [reply]
Awesome, very helpful suggestions. Thanks very much for your reply and for this great tool!

**Reply by kevlarr** [reply]
> The WSL filesystem isn't directly visible on the Windows host, so that isn't possible.

Out of curiosity, how are other Windows programs able to access those files? Eg. if I create a shortcut `C:\Users\<user>\wezterm.lua` that targets `\\wsl.localhost\Ubuntu-22.04\home\<user>\.config\wezterm\wezterm.lua`, I can directly open that shortcut in VSCode and even make changes to it, but trying to point to it like `wezterm-gui.exe --config-file="C:\Users\<user>\wezterm.lua"` can't access the garget.

(Probably obvious but I have practically zero understanding of how WSL1/2 works. And also, WezTerm is awesome! I wish I'd discovered it awhile ago. Thanks for making such an awesome - and easy to configure - cross-platform term emulator.)
